mavutil.py

```diff
@@ -898,15 +902,25 @@ def recv(self,n=None):
             if e.errno in [ errno.EAGAIN, errno.EWOULDBLOCK, errno.ECONNREFUSED ]:
                 return ""
             raise
-        if self.udp_server or self.broadcast:
+        if self.udp_server:
+            self.clients.add(new_addr)
+            self.clients_last_alive[new_addr] = time.time()
+        elif self.broadcast:
             self.last_address = new_addr
```

Comment:
Should we do the same thing for broadcast ?

Reply:
has this been resolved?

Reply:
I don't think we need it right now. There's some additional logic in the broadcast section I didn't quite understand the reasoning behind, too.